src/algebra/order/with_zero.lean

```diff
@@ -107,6 +107,9 @@ lemma zero_lt_iff : 0 < a ↔ a ≠ 0 :=
 lemma ne_zero_of_lt (h : b < a) : a ≠ 0 :=
 λ h1, not_lt_zero' $ show b < 0, from h1 ▸ h
 
+/-- Same as ne_zero_of_lt, except it takes an explicit argument. -/
+lemma ne_zero_of_lt' (b' : α) [fact (b' < a)] : a ≠ 0 := @ne_zero_of_lt _ _ b' _ (fact.out _)
+
```

Comment:
I don't really understand why this is useful

Reply:
I will remove this for now and add it when needed.